AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -350,25 +350,69 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
                 return
             }
 
-            storageAdapter.delete(untypedModelType: modelType,
-                                  modelSchema: self.modelSchema,
-                                  withIdentifier: remoteModel.model.identifier(schema: self.modelSchema),
-                                  condition: nil) { response in
-                switch response {
+            storageAdapter.query(modelSchema: self.modelSchema,
+                                 predicate: remoteModel.model.identifier(schema: self.modelSchema).predicate,
+                                 eagerLoad: false) { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case .success(let models):
+                    if let existingModel = models.first {
+                        self.delete(
+                            modelType: modelType,
+                            storageAdapter: storageAdapter,
+                            remoteModel: remoteModel,
+                            existingModel: existingModel,
+                            promise: promise)
+                    } else {
+                        promise(.failure(
+                            .nonUniqueResult(model: modelType.modelName,
+                                             count: models.count)))
+                    }
```

Comment:
i think if we want to go this route, this should return success: `promise(.success(.applied( ?)` so we move onto saving the MutationSync's metadata